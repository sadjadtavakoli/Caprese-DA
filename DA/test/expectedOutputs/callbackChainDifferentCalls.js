
#3 function 3 entered from 1
#3 function 3 exited to function 1
#21 function 21 entered from 1
#21 function 21 exited to function 1
#7 function 7 entered from 3
#9 function 9 entered from 7
#9 function 9 exited to function 7
#9 function 9 entered from 7
#9 function 9 exited to function 7
#9 function 9 entered from 7
#9 function 9 exited to function 7
#9 function 9 entered from 7
#9 function 9 exited to function 7
#7 function 7 exited to function 3
#14 function 14 entered from 3
#14 function 14 exited to function 3
#25 function 25 entered from 21
#25 function 25 exited to function 21
#end Execution