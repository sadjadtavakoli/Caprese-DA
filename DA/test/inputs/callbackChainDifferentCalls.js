let fs = require('fs')

fs.readdir('./test/inputs', function (err, files) {
    if (err) {
        console.log('Error finding files: ' + err)
    } else {

        let foo = function() {
            console.log("Done!");
        }

        let filename = "simpleTrace.js"
        fs.readFile('./test/inputs/' + filename, 'utf8', (err, data) => {
            if (err) {
                console.error(err)
                return
            } else {
                let items = [1]
                items.forEach(function (item) {
                    // Do nothing
                })
            }
        });
        setTimeout(foo, 10 * 1000)

        filename = "simpleCallback.js";
        fs.readFile('./test/inputs/' + filename, 'utf8', (err, data) => {
            if (err) {
                throw err
            } else {
                    // Do nothing
            }
        });
    }
});