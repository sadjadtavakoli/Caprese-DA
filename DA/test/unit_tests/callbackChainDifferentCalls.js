let fs = require('fs')

fs.readdir('./test/inputs', function (err, files) {
    if (!err) {

        let filename = "callChain.js"
        fs.readFile('./test/unit_tests/' + filename, 'utf8', (error, data) => {
            let items = [1, 2, 3, 4]
            items.forEach(function (item) {
                console.log("inside foo loop")
            })
        });

        filename = "callChain.js";
        fs.readFile('./test/unit_tests/' + filename, 'utf8', (err2, data) => {
            console.log("inside second one")

        });
    }
});


fs.readdir('./test/inputs', function (err, files) {
    if (!err) {

        let filename = "callChain.js"
        fs.readFile('./test/unit_tests/' + filename, 'utf8', (error, data) => {});
    }
});