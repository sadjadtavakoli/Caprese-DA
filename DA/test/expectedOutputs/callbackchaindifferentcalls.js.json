{"keyMap":{"t_1":"callbackchaindifferentcalls.js-test/unit_tests/callbackchaindifferentcalls.js-1-28","t_31":"arrowAnonymousFunction-test/unit_tests/callbackchaindifferentcalls.js-3-19","t_41":"arrowAnonymousFunction-test/unit_tests/callbackchaindifferentcalls.js-22-28","t_112":"arrowAnonymousFunction-test/unit_tests/callbackchaindifferentcalls.js-26-26","t_72":"arrowAnonymousFunction-test/unit_tests/callbackchaindifferentcalls.js-7-12","t_160":"arrowAnonymousFunction-test/unit_tests/callbackchaindifferentcalls.js-9-11","t_82":"arrowAnonymousFunction-test/unit_tests/callbackchaindifferentcalls.js-15-17"}}